FEAT(#330): ADD GODFATHER FORM

IsmaelP19 opened the pull request:
In this pr I created the goodfather form. 

### ⚠️ Important!
**Be sure to create a superuser first. You will have to create a NGO if none is created yet.**


You can play the next scenarios to check the form functionality:

- [x] You can navigate to the goodfather list. It should be empty now.
- [x] You can navigate to the goodfather creation form. To do so, please, go to http://localhost:8000/user/goodfather/create or just click on the "add child" button.
- [x] If you haven't logged in, it will redirect you to the login page and after have logged in, it will check if the ngo of the user is "Videssur" (doesn't matter if it's uppercase or lowercase). If not, it will redirect to http://localhost:8000/
- [x] Check if colors are correct.
- [x] Now, try the form on your own. Make as much tests as you want! 😉️

### Disclaimer: you cannot edit a goodfather yet because the functionality is being reviewed by the backend side on #305

ezegonmac commented on person/views.py:
```diff
@@ -287,7 +287,7 @@ def godfather_update(request,godfather_slug):
         else:
             messages.error(request, 'Formulario con errores')
-    return render(request, 'godfather_form.html', {"form": form})
+    return render(request, 'person/godfather/godfather_form.html', {"form": form})
```
I think we are naming the form files as "form.html" only, maybe I am wrong but I have been doing like this 😳

IsmaelP19 replied:
Okay, I will rename it. I just changed the path because it was on `/templates`, instead of `/templates/person/godfather`

ezegonmac commented on person/models.py:
```diff
@@ -206,7 +206,7 @@ class GodFather(Person):
         max_length=100, verbose_name='Titular de cuenta bancaria')
     bank_account_reference = models.CharField(
-        max_length=100, verbose_name='Referencia de cuenta bancaria', validators=[RegexValidator(r'^[0-9]+$')])
+        max_length=100, verbose_name='Referencia de cuenta bancaria', validators=[RegexValidator(r'^[0-9]+$')]) #for example, 1234567890
```
Great addition 👍🏻

ezegonmac commented on person/views.py:
```diff
@@ -248,7 +248,9 @@ def godfather_create(request):
-    return render(request, 'godfather_form.html', {"form": form, "title": "Añadir Padrino"})
+
+    form = CreateNewGodFather()
+    return render(request, 'person/godfather/godfather_form.html', {"form": form, "title": "Añadir Padrino"})
```
You have forgotten to change "godfather_form.html" to "form.html"

IsmaelP19 replied:
Ups 🙃

ezegonmac reviewed (changes requested):
In addition to what has already been mentioned in the comments (the mayority of errors from the merge conflicts you can fix them easily), I have two doubts:

1. Should name and surname be required, I think they are the most important
2. Should bank account info be required? I think no, becouse if you pay with cash ("efectivo") you dont need to add a bank account

IsmaelP19 commented on person/views.py:
```diff
@@ -248,7 +248,9 @@ def godfather_create(request):
             messages.error(request, 'Formulario con errores')
-    return render(request, 'godfather_form.html', {"form": form, "title": "Añadir Padrino"})
+
+    form = CreateNewGodFather()
```
Ups 🙃 , should be fixed now 👍🏼

IsmaelP19 commented:
> In addition to what has already been mentioned in the comments (the mayority of errors from the merge conflicts you can fix them easily), I have two doubts:
> 
> 1. Should name and surname be required, I think they are the most important
> 2. Should bank account info be required? I think no, becouse if you pay with cash ("efectivo") you dont need to add a bank account


I appreciate your review.
As for your concerns, **both** have in common that **should be addressed by the backend team**. I could add a specific validator in the frontend to only allow the submit if name and surnames were provided, however, I would have to do this on every form and it  would be easier to change the Person model, which is the parent model to many of the entities that have this problem, like Godfather or even Child. 
As for number two, I could also add a script on the frontend to let the user send the form with no bank information on it if the pay method selected is cash, but this would result in an error, because in the backend they have specified that bank information must be provided no matter what. 

I agree with you in the aspects you said, and I think that if the backend team address these issues, all the form views would be fixed inmediately (hopefully). Maybe, with bank information we would have to do something else, but I honestly don't think so.

I hope you don't find any more errors, but please, if you do so, just let me know and I will fix it.

_PS: I also added a validator that if termination date was provided, it should be after the start date. I added this validation in both godfather and home form._

ezegonmac commented:
Great, all changes have been added correctly! I will adress those problems to backend teams 👍🏻

ezegonmac reviewed (approved):
Great, all changes have been added correctly! I will adress those problems to backend teams 👍🏻
